seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java

```diff
@@ -36,5 +36,10 @@ public enum SqlType {
     TIME,
     TIMESTAMP,
     ROW,
+    BINARY_VECTOR,
+    FLOAT_VECTOR,
+    FLOAT16_VECTOR,
+    BFLOAT16_VECTOR,
+    SPARSE_FLOAT_VECTOR,
```

Comment:
Please share some information about why we should add these new type and what's used for?  We should be careful about add new type. cc @hailin0 @EricJoy2048

Reply:
Vector database need support more vectorType

Reply:
Because vectors are just display as arrays, but the management and processing methods are different from arrays.
For example, arrays have no dimension restrictions, while the data in vectors has fixed dimensions.
If the array type is used, in the future, if functions are expanded or data is verified, special processing must be performed for the vector situation, which will make the code look very strange and not decoupled, so the vector type is added.
In addition, many vector databases on the market have separate types for vector columns, and arrays and vectors do not share the same type in terms of external SQL types.